scripts/ImportExportTest.sh

```diff
@@ -0,0 +1,270 @@
+#!/usr/bin/env bash
+
+set -euo pipefail
+IMPORT_TEST_TYPE="${1}"
+
+# Bash script to test the import/exports.
+# ast import/export tests:
+#   - first exporting a .sol file to JSON, then loading it into the compiler
+#     and exporting it again. The second JSON should be identical to the first.
+# evm-assembly import/export tests:
+#   - <TODO>
+
+READLINK=readlink
+if [[ "$OSTYPE" == "darwin"* ]]; then
+    READLINK=greadlink
+fi
+REPO_ROOT=$(${READLINK} -f "$(dirname "$0")"/..)
+SOLIDITY_BUILD_DIR=${SOLIDITY_BUILD_DIR:-${REPO_ROOT}/build}
+SOLC=${SOLIDITY_BUILD_DIR}/solc/solc
+SPLITSOURCES=${REPO_ROOT}/scripts/splitSources.py
+
+# shellcheck disable=SC1090,SC1091
+source "${REPO_ROOT}/scripts/common.sh"
+
+SYNTAXTESTS_DIR="${REPO_ROOT}/test/libsolidity/syntaxTests"
+SEMANTICTESTS_DIR="${REPO_ROOT}/test/libsolidity/semanticTests"
+ASTJSONTESTS_DIR="${REPO_ROOT}/test/libsolidity/ASTJSON"
+
+FAILED=0
+UNCOMPILABLE=0
+TESTED=0
+
+if [[ "$(find . -maxdepth 0 -type d -empty)" == "" ]]; then
+    fail "Test directory not empty. Skipping!"
+fi
+
+function Ast_ImportExportEquivalence
+{
+    local nth_input_file="$1"
+    IFS=" " read -r -a all_input_files <<< "$2"
+
+    # save exported json as expected result (silently)
+    $SOLC --combined-json ast --pretty-json --json-indent 4 "$nth_input_file" "${all_input_files[@]}" > expected.json 2> /dev/null
+    # import it, and export it again as obtained result (silently)
+    if ! $SOLC --import-ast --combined-json ast --pretty-json --json-indent 4 expected.json > obtained.json 2> stderr.txt
+    then
+        # For investigating, use exit 1 here so the script stops at the
+        # first failing test
+        # exit 1
+        FAILED=$((FAILED + 1))
+        printError -e "ERROR: AST reimport failed for input file $nth_input_file"
+        printError
+        printError "Compiler stderr:"
+        cat ./stderr.txt >&2
+        printError
+        printError "Compiler stdout:"
+        cat ./obtained.json >&2
+        return 1
+    fi
+    set +e
+    diff_files expected.json obtained.json
+    DIFF=$?
+    set -euo pipefail
```

Comment:
hmm.. maybe its just ok to do set -e here.. not sure whether the other settings saved. @cameel do you know?

Reply:
Yeah, just `+e`/`-e` is probably enough.

Reply:
But you can avoid this pattern completely with something like
```bash
if ! diff_files expected.json obtained.json
```